pkg/api/rest/update_test.go

```diff
@@ -16,95 +16,4 @@ limitations under the License.
 
 package rest
 
-import (
-	"testing"
-
-	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/runtime"
-	"k8s.io/kubernetes/pkg/util"
-)
-
-func makeValidService() api.Service {
-	return api.Service{
-		ObjectMeta: api.ObjectMeta{
-			Name:            "valid",
-			Namespace:       "default",
-			Labels:          map[string]string{},
-			Annotations:     map[string]string{},
-			ResourceVersion: "1",
-		},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"key": "val"},
-			SessionAffinity: "None",
-			Type:            api.ServiceTypeClusterIP,
-			Ports:           []api.ServicePort{{Name: "p", Protocol: "TCP", Port: 8675, TargetPort: util.NewIntOrStringFromInt(8675)}},
-		},
-	}
-}
-
-// TODO: This should be done on types that are not part of our API
-func TestBeforeUpdate(t *testing.T) {
-	testCases := []struct {
-		name      string
-		tweakSvc  func(oldSvc, newSvc *api.Service) // given basic valid services, each test case can customize them
-		expectErr bool
-	}{
-		{
-			name: "no change",
-			tweakSvc: func(oldSvc, newSvc *api.Service) {
-				// nothing
-			},
-			expectErr: false,
-		},
-		{
-			name: "change port",
-			tweakSvc: func(oldSvc, newSvc *api.Service) {
-				newSvc.Spec.Ports[0].Port++
-			},
-			expectErr: false,
-		},
-		{
-			name: "bad namespace",
-			tweakSvc: func(oldSvc, newSvc *api.Service) {
-				newSvc.Namespace = "#$%%invalid"
-			},
-			expectErr: true,
-		},
-		{
-			name: "change name",
-			tweakSvc: func(oldSvc, newSvc *api.Service) {
-				newSvc.Name += "2"
-			},
-			expectErr: true,
-		},
-		{
-			name: "change ClusterIP",
-			tweakSvc: func(oldSvc, newSvc *api.Service) {
-				oldSvc.Spec.ClusterIP = "1.2.3.4"
-				newSvc.Spec.ClusterIP = "4.3.2.1"
-			},
-			expectErr: true,
-		},
-		{
-			name: "change selectpor",
-			tweakSvc: func(oldSvc, newSvc *api.Service) {
-				newSvc.Spec.Selector = map[string]string{"newkey": "newvalue"}
-			},
-			expectErr: false,
-		},
-	}
-
-	for _, tc := range testCases {
-		oldSvc := makeValidService()
-		newSvc := makeValidService()
-		tc.tweakSvc(&oldSvc, &newSvc)
-		ctx := api.NewDefaultContext()
-		err := BeforeUpdate(Services, ctx, runtime.Object(&oldSvc), runtime.Object(&newSvc))
-		if tc.expectErr && err == nil {
-			t.Errorf("unexpected non-error for %q", tc.name)
-		}
-		if !tc.expectErr && err != nil {
-			t.Errorf("unexpected error for %q: %v", tc.name, err)
-		}
-	}
-}
+// The original "TestBeforeUpdate" test depends on service strategy, is moved to the registry/service package.
```

Comment:
Yeah, sorry, I didn't realize you removed all the tests from these files-- just remove the file in that case. The git history will tell us what happened.